mitiq/interface/mitiq_qiskit/conversions.py

```diff
@@ -133,45 +133,42 @@ def _add_identity_to_idle(
     Note: An idle qubit is a qubit without any gates (including Qiskit
         barriers) acting on it.
     """
+    all_qubits = set(circuit.qubits)
+    used_qubits = set()
+    idle_qubits = set()
+    # Get used qubits
+    for op in circuit.data:
+        _, qubits, _ = op
+        used_qubits.update(set(qubits))
+    idle_qubits = all_qubits - used_qubits
+    # Modify input circuit applying I to idle qubits
+    for q in idle_qubits:
+        circuit.i(q)
 
-    data = copy.deepcopy(circuit._data)
-    bit_indices = set()
-    idle_bit_indices = set()
-    for op in data:
-        gate, qubits, cbits = op
-        bit_indices.update(set(bit.index for bit in qubits))
-    for index in range(circuit.num_qubits):
-        if index not in bit_indices:
-            idle_bit_indices.add(index)
-            circuit.i(index)
-    return idle_bit_indices
+    return idle_qubits
 
 
 def _remove_identity_from_idle(
     circuit: qiskit.QuantumCircuit,
-    idle_indices: Set[int],
+    idle_qubits: Set[qiskit.circuit.Qubit],
 ) -> None:
-    """Removes identities from the circuit corresponding to the set of indices.
+    """Removes identities from the circuit corresponding to the input
+    idle qubits.
     Used in conjunction with _add_identity_to_idle to preserve idle qubits and
     indices in conversion.
```

Comment:
```suggestion
    Used in conjunction with _add_identity_to_idle to preserve idle qubits in
    conversion.
```

That's not true anymore, right?